### PR TITLE
Ransomware exploitation details report service

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do? 

#1240 

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing? 
* [x] ~Was the CHANGELOG.md updated to reflect the changes?~
* [x] ~Was the documentation framework updated to reflect the changes?~

## Testing Checklist

* [x] ~Added relevant unit tests?~
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running the Island and logging response from the API endpoint 
* [x] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

File: monkey/monkey_island/cc/resources/ransomware_report.py [hunk 1]
```diff
@@ -2,9 +2,12 @@
...
 from monkey_island.cc.resources.auth.auth import jwt_required
+from monkey_island.cc.services.ransomware_report import RansomwareReportService
 
 
 class RansomwareReport(flask_restful.Resource):
     @jwt_required
     def get(self):
-        return jsonify({"report": None})
+        return jsonify(
+            {"report": None, "stats": RansomwareReportService.get_exploitation_details()}
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/resources/ransomware_report.py:12]:
```suggestion
            {"report": None, "propagation": RansomwareReportService.get_exploitation_details()}
```

File: monkey/monkey_island/cc/services/ransomware_report.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from monkey_island.cc.services.reporting.report import ReportService
+
+
+class RansomwareReportService:
+    def __init__(self):
+        pass
+
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/services/ransomware_report.py:7]: This code is unnecessary.

File: monkey/monkey_island/cc/services/ransomware_report.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+from monkey_island.cc.services.reporting.report import ReportService
+
+
+class RansomwareReportService:
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/services/ransomware_report.py:4]: I realize this is the pattern for the rest of the services, but a class that stores no state and only contains static methods doesn't make a whole lot of sense. The class doesn't serve any purpose. This pattern seems like something someone picked up from Java and carried over to Python.

We can just make `get_exploitation_details()` a function and call `ransomware_report.get_exploitation_details()`

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/monkey_island/cc/services/ransomware_report.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+from monkey_island.cc.services.reporting.report import ReportService
+
+
+def get_exploitation_details():
+    scanned = ReportService.get_scanned()
+    exploited = ReportService.get_exploited()
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/services/ransomware_report.py:6]: What exactly do we need for statistics? There's no need to send all exploited and especially scanned nodes. Maybe the count is enough?

Comment (shreyamalviya) [reply monkey/monkey_island/cc/services/ransomware_report.py:6]: Won't we need the details for the table (we need the machine, how it was exploited, and ransomware simulation details)?

**Review by ilija-lazoroski — APPROVED** [review]

File: monkey/monkey_island/cc/services/ransomware_report.py (hunk 4 above)
Comment (mssalvatore) [reply monkey/monkey_island/cc/services/ransomware_report.py:6]: I think the front end does very little (if any) report processing. So we could do all of the processing on the back end and send just the information that will be displayed.

```python
{"numScanned":  len(scanned), "numExploited": len(exploited), "exploits" {"wmi": 7, "smb": 13 ... }}
```

Comment (shreyamalviya) [reply monkey/monkey_island/cc/services/ransomware_report.py:6]: So the ransomware report won't show the exploit that gave access successful to a particular machine?

Comment (shreyamalviya) [reply monkey/monkey_island/cc/services/ransomware_report.py:8]: Done
image

File: monkey/monkey_island/cc/services/ransomware_report.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+
+def get_propagation_stats() -> Dict:
+    scanned = ReportService.get_scanned()
+    exploited = ReportService.get_exploited()
+
+    return {
+        "num_scanned_nodes": len(scanned),
+        "num_exploited_nodes": len(exploited),
+        "count_per_exploit": _get_exploit_counts(exploited),
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/services/ransomware_report.py:13]: It seems weird to have two of these named "num" and one named "count". "num" and "count" basically mean the same thing, and I don't think the distinction here is valuable for the consumer of the API.

I think "num_per_exploit", "num_hosts_per_exploit", or "num_nodes_per_exploit" would be better.

I'm also wondering if "hosts" would be better than "nodes", but I don't feel too strongly about this.

**Review by mssalvatore — CHANGES_REQUESTED** [review]

**Pull request merged**